### ISSUE TITLE
free-jqgrid.d.ts is not compilled in VS due to /// <reference types="jquery" />

### ISSUE THREAD

**Issue by askrinnik** [free-jqgrid/jqGrid#369]
Hi Oleg

Visual Studio 2017 (15.3.4) 
MVC5 web project with TypeScript.
jquery.TypeScript.DefinitelyType and jqueryui.TypeScript.DefinitelyTyped were loaded from NuGet.
It places *.d.ts files to folder Scripts\typing\\**\\*.d.ts
jqGrid was loaded from NuGet

**Compilation errors:**

> Build:Cannot find type definition file for 'jquery'.	\Scripts\free-jqGrid\ts\free-jqgrid.d.ts	8	
> Build:Cannot find type definition file for 'jqueryui'.	\Scripts\free-jqGrid\ts\free-jqgrid.d.ts	9	

It caused because 
`/// <reference types="jquery" /> `
requires that *.d.ts files should be placed into @types/**/*.d.ts, but files ar in different folder

**Workaround:**
Replace into \Scripts\free-jqGrid\ts\free-jqgrid.d.ts
```
/// <reference types="jquery" />
/// <reference types="jqueryui" />
```
to 
```
/// <reference path="../../typings/jquery/jquery.d.ts" />
/// <reference path="../../typings/jqueryui/jqueryui.d.ts" />
```

Oleg, do you have any thought how to fix the issue without making changes in free-jqgrid.d.ts file?

**Comment by OlegKi** [comment]
Thank you for the post! The problem exist only in case of using free jqGrid from NuGet. The problem, which you described is the same like in [the issue #353](https://github.com/free-jqgrid/jqGrid/issues/353). I described the workaround [here](https://github.com/free-jqgrid/jqGrid/issues/353#issuecomment-323547393). Alternatively one can just remove `free-jqgrid.d.ts` from Visual Studio package if one don't use free jqGrid from TypeScript. I plan to modify the lines
```TypeScript
/// <reference types="jquery" />
/// <reference types="jqueryui" />
```
to
```TypeScript
/// <reference path="../../typings/jquery/jquery.d.ts" />
/// <reference path="../../typings/jqueryui/jqueryui.d.ts" />
```
but **only for the `free-jqgrid.d.ts` included in NuGet package** and not in the `free-jqgrid.d.ts` included in npm package.

Sorry, I just didn't found time to make some small changes in the code of free jqGrid and to publish the current version from GitHub as new version 4.15.0, where the NuGet package will be fixed. I'll try to do this very soon.

**Comment by askrinnik** [comment]
Sounds great!
Thanks a lot for your afford.
Best wishes

**Comment by OlegKi** [comment]
I published just now version 4.15.0 and the new [NuGet package](https://www.nuget.org/packages/free-jqGrid/4.15.0). It contains the changes described above. Please try it. I hope that the issue can be closed now.

**Comment by askrinnik** [comment]
Excellent. 
Thanks a lot!

Best regards
